gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireSparkConnectorCacheImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.gemstone.gemfire.internal.cache;
+
+import com.gemstone.gemfire.cache.AttributesFactory;
+import com.gemstone.gemfire.cache.DataPolicy;
+import com.gemstone.gemfire.cache.client.ClientRegionShortcut;
+import com.gemstone.gemfire.cache.client.Pool;
+import com.gemstone.gemfire.cache.client.PoolFactory;
+import com.gemstone.gemfire.cache.query.QueryService;
+import com.gemstone.gemfire.distributed.DistributedSystem;
+
+/**
+ * Created by ashahid on 3/30/17.
+ */
+public class GemFireSparkConnectorCacheImpl extends GemFireCacheImpl {
+
+  public GemFireSparkConnectorCacheImpl(PoolFactory pf, DistributedSystem system, CacheConfig cacheConfig) {
+    super(false, pf, system, cacheConfig);
+  }
+
+  public static GemFireCacheImpl create(PoolFactory pf, DistributedSystem system,
+      CacheConfig cacheConfig) {
+    return new GemFireSparkConnectorCacheImpl(pf, system, cacheConfig).init();
+  }
+
+  @Override
+  protected GemFireCacheImpl init() {
+    PoolFactory temp = this.clientpf;
+    super.init();
+    //re assign the clientPf which would otherwise be nullified by the super call
+    this.clientpf = temp;
+
+    this.determineDefaultPool();
+    AttributesFactory af = new AttributesFactory();
+    af.setDataPolicy(DataPolicy.EMPTY);
+    UserSpecifiedRegionAttributes ra = (UserSpecifiedRegionAttributes)af.create();
+    ra.requiresPoolName = true;
+    this.setRegionAttributes(ClientRegionShortcut.PROXY.toString(), ra);
```

Comment:
Just to understand.
So against this id 'ClientRegionShortcut.PROXY.toString()' we create a regionattribute object and store in the cache's map. Probably this will be used by the client regions on this vm. Would we require any other attribute? The region shortcut in the cache xml will have a set of attributes against this id. So the final region attribute will be DataPolicy.NORMAL + whatever is mentioned in the region shortcut against this id?